org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -20,10 +20,52 @@ public class NearInteractionModeDetector : ProximityDetector
         [Tooltip("The set of near interactors that belongs to near interaction")]
         private List<XRBaseInteractor> nearInteractors;
 
+        /// <summary>
+        /// Used to keep track of the previously detected colliders so that we can know which
+        /// colliders stopped being detected and update their buttons front plate and rounded
+        /// rect if they are labeled as dynamic (based on proximity).
+        /// </summary>
+        List<Collider> previouslyDetectedColliders = new List<Collider>();
+
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
-            return base.IsModeDetected() || IsNearInteractorSelecting();
+            bool result = base.IsModeDetected() || IsNearInteractorSelecting();
+            if (result)
+            {
+                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                {
+                    Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
+                    if (!DetectedColliders.Contains(previouslyDetectedCollider) && previouslyDetectedCollider != null)
+                    {
+                        IXRProximityInteractable nearInteractionMode = previouslyDetectedCollider.GetComponent<IXRProximityInteractable>();
+                        if (nearInteractionMode != null)
```

Comment:
Thank you for the feedback and additional explanation.  I updated the code to use TryGetComponent instead of GetComponent in https://github.com/ms-RistoRK/MixedRealityToolkit-Unity/commit/8ba56f4008476f7a76a02d552321ec002b9ad181